File: absoluteValuesSumMinization/absoluteValuesSumMinization.ts
function absoluteValuesSumMinization(a : number[]) : number{
    const isEven = a.length % 2 == 0;
    return isEven? a[a.length / 2 - 1] : a[Math.floor(a.length/2)];
}

console.log(absoluteValuesSumMinization([2,4,7]));
console.log(absoluteValuesSumMinization([2,4,7,8]));
console.log(absoluteValuesSumMinization([2,3,3.5,4,5,7]));

File: absoluteValuesSumMinization/absoluteValuesSumMinization.java
// package absoluteValuesSumMinization;

public class absoluteValuesSumMinization{

    public static int absoluteValuesSum(int [] a){
        boolean isEven = a.length % 2 == 0;
        int len = a.length / 2;
        int l = (int)Math.floor(len);
        return isEven? a[len - 1] : a[l];
    }

    public static void main(String[] args) {
        int arr[] = {2,4,7};
        int hello = absoluteValuesSum(arr);
        System.out.println(hello);
    }
}
